### PR TITLE
PR: Add a Shortcuts Summary window

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes: #3275

Same as #3331 but  to the 3.x branch, so It could be merged early

### CODE REVIEW

**Comment by ccordoba12** [comment]
@rlaverde, let's have first the work to remove `fixed_shortcut` before merging this PR :-)

**Comment by ccordoba12** [comment]
@rlaverde, could you rebase or merge this with the 3.x branch? It seems you started this work before 3.0 was released, and that's making Travis to fail :-)

**Comment by goanpeca** [comment]
@rlaverde could you paste a screenshot just for reference?

Is it finished otherwise?

**Comment by rlaverde** [comment]
@goanpeca This isn't ready, I'm planning to rewrite some of the logic to calculate the maximum amount of lines for column

**Comment by jitseniesen** [comment]
Sorry, I have been rather picky 😊

**Comment by ccordoba12** [comment]
Besides, the removal of fixed shortcuts have to be merged first :-)

File: spyder/widgets/shortcuts.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+# Local imports
+from spyder.config.base import _
+from spyder.config.gui import iter_shortcuts
+
+# Constants
+MAX_FONT_SIZE = 16
+MIN_FONT_SIZE = 8
+
+class ShortCutsSummaryDialog(QDialog):
```
Comment (ccordoba12) [inline spyder/widgets/shortcuts.py]: Please name this as ShortcutsSummaryDialog

**Comment by ccordoba12** [comment]
@rlaverde, please finish this one this week. After PR #3498, all our shortcuts should appear in this dialog :-)

File: spyder/widgets/shortcuts.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
```
Comment (ccordoba12) [inline spyder/widgets/shortcuts.py]: Please name this file as `shortcutssummary.py` because just `shortcuts` is a bit vague :-)

**Comment by ccordoba12** [comment]
Tests are failing when they run this line

    python spyder/widgets/shortcutssummary.py

**Comment by ccordoba12** [comment]
@rlaverde, please solve conflicts for this one and finish it by the end of the week.

**Comment by rlaverde** [comment]
I tried to dynamically update the dialog to avoid it to be bigger of the screen, but it didn't work, because the size of the layout isn't update when adding elements :/

I leave the "smart" logic that try to calculate the best amount of elements by column,  and added an `QScrollArea` to prevent that the window will be bigger than the screen

**Comment by ccordoba12** [comment]
Please rebase on top of 3.x and post a screenshot of how things look now.

**Comment by ccordoba12** [comment]
The rebase is needed to fix the failures in Circle and Travis.

**Comment by goanpeca** [comment]
Screenshot???? @rlaverde

**Comment by rlaverde** [comment]
shortcuts summary

**Comment by ccordoba12** [comment]
I like it, a lot!! But I don't know if showing this dialog to the full width of the screen is the best option.

@goanpeca, @jitseniesen, what do you think?

**Comment by jitseniesen** [comment]
> But I don't know if showing this dialog to the full width of the screen is the best option.

Full width seems okay to me because the shortcuts won't fit otherwise. Unless we select and display only the "important" shortcuts. 

I guess I would prefer it if it scrolled vertically instead of horizontally.

**Comment by ccordoba12** [comment]
> I guess I would prefer it if it scrolled vertically instead of horizontally.

Yep, me too, but I don't know how we could achieve this.

Ok, let's merge this because it's a great improvement. We can improve it later, as you said :-)

**Comment by goanpeca** [comment]
@rlaverde its `Shortcut` not `ShortCut`, why the camel casing?

**Pull request merged**